### PR TITLE
Add support for argon2id

### LINKED ISSUE #1734: LDAP password import failing (argon2)
### I did this

Trying to use the LDAP importer.

```
ERROR    │     ┝━ 🚨 [error]: password_import was unable to convert hash format | hint: {ARG | entry_id: mmmmmmm
ERROR    │     ┝━ 🚨 [error]: Pre-Modify operation failed (plugin), Plugin(CredImport("password_import was unable to convert hash format")) | event_tag_id: 1
ERROR    │     ┕━ 🚨 [error]: Unable to apply modifications to sync entries.
ERROR    ┕━ 🚨 [error]: Internal error -> Response sent | event_tag_id: 4 | status: 500 - Internal Server Error
```

The hash uses argon2id.

### I expected the following

Successful import of users and groups, reusing the given hashes, since they are compatible with `crypt()`.

### Kanidm version (and git commit)

[`780c82f` (#1733)](https://github.com/kanidm/kanidm/pull/1733/commits/780c82f8ec1b40919aa16aec8524af070fb89d47)

### Operating System / Version

Linux sso2 5.15.113 #1-NixOS SMP Wed May 24 16:36:55 UTC 2023 x86_64 GNU/Linux

### PR DESCRIPTION
**Pull request by Firstyear**
Fixes #1734 - This adds support for argon2id. It also allows auto-tuning it's parameters based on system performance. I am still unsure yet if to change to it by default, as I want to consult some members of the SUSE product security team first. 

- [ x ] This pr contains no AI generated code
- [ x ] cargo fmt has been run
- [ x ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Comment by Firstyear** [comment]
cc @mweinelt

**Comment by Firstyear** [comment]
@mweinelt I'm assuming this worked for you?

File: libs/crypto/src/lib.rs [hunk 1]
```diff
@@ -84,31 +116,37 @@ impl fmt::Debug for DbPasswordV1 {
...
         CryptoPolicy {
             pbkdf2_cost: PBKDF2_MIN_NIST_COST,
+            argon2id_params: Params::default(),
         }
     }
 
-    pub fn time_target(t: Duration) -> Self {
-        let r = match Password::bench_pbkdf2(PBKDF2_MIN_NIST_COST * 10) {
+    pub fn time_target(target_time: Duration) -> Self {
+        const PBKDF2_BENCH_FACTOR: usize = 10;
```
Comment (yaleman) [inline libs/crypto/src/lib.rs:133]: Wouldn't this const be better up with the other things? 😄

**Review by yaleman — APPROVED** [review]
More algos moar better.

File: libs/crypto/src/lib.rs (hunk 1 above)
Comment (Firstyear) [reply libs/crypto/src/lib.rs:133]: It could be, but it's also only ever used in this one function so I think tho locality is nicer

**Comment by mweinelt** [comment]
> @mweinelt I'm assuming this worked for you? 

Yes, it did. Thanks again

**Pull request merged**